src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -152,23 +168,44 @@ export class CourseManagementService {
         if (userRefresh) {
             params = params.set('refresh', String(true));
         }
-        return this.http.get<Course>(`${this.resourceUrl}/${courseId}/for-dashboard`, { params, observe: 'response' }).pipe(
+        return this.http.get<CourseForDashboardDTO>(`${this.resourceUrl}/${courseId}/for-dashboard`, { params, observe: 'response' }).pipe(
+            map((res: HttpResponse<CourseForDashboardDTO>) => {
+                if (res.body) {
+                    const courseForDashboardDTO: CourseForDashboardDTO = res.body;
+                    this.saveScoresInStorage(courseForDashboardDTO);
```

Comment:
Why don't we push the course into the course-storage here, too, as we're getting it freshly?

Reply:
Some fields in the Course are set in the `processCourseEntityResponseType()` method further in the chain before the course is saved in the course storage. This `processCourseEntityResponseType()` method expects a Course object not a CourseForDashboardDTO.

Reply:
I have missed your ``tap`` operation below. Sorry.